### DISCUSSION TITLE
Issue with C++ Template Classes (clang: No template named)

### DISCUSSION THREAD

**Discussion by mbucataru** [zed-industries/zed#7739] [Feature Requests]
I'm running into an issue when using Zed in C++. I have a List.h file with the declarations template class List and a List.hpp file with the definitions. This .hpp file is included at the bottom of the .h file.

However, using Zed, I'm getting a clang error where it says that in my .hpp file, "No template named 'List'"

This problem does not happen if I place the code from the .hpp file right into the slot where the #include "List.hpp" was. So, it seems to be an issue with the file being able to recognize other files from the same directory, or I am simply misunderstanding something. Thank you for the help!

<img width="717" alt="Screenshot 2024-02-13 at 1 30 42 PM" src="https://github.com/zed-industries/zed/assets/120226852/9b398aec-c232-4b26-84d1-401bf9c8382c">
<img width="340" alt="Screenshot 2024-02-13 at 1 31 21 PM" src="https://github.com/zed-industries/zed/assets/120226852/01fbc9ce-a42e-4907-9974-8b0a46a6a7b1">
<img width="547" alt="Screenshot 2024-02-13 at 1 32 14 PM" src="https://github.com/zed-industries/zed/assets/120226852/9e675453-1a74-42f0-bc5b-f10680f74003">
<img width="565" alt="Screenshot 2024-02-13 at 1 33 02 PM" src="https://github.com/zed-industries/zed/assets/120226852/5cf10c4a-4d22-41be-b93b-7f4e3416df5d">

**Comment by adult-zachary** [comment]
I am also having this issue. Did you ever find a solution?

**Comment by JosephTLyons** [comment]
Thanks for reaching out. We want to better support our community, and the best way for us to do that is through our official Discord. The team is active there and happy to help.

We’re closing the “Help and General Q&A” category on GitHub Discussions and directing all support questions to Discord going forward.

https://zed.dev/support
